model/github/github_repository.rb

```diff
@@ -74,6 +74,10 @@ def setup_blob_storage
       rescue Aws::S3::Errors::BucketAlreadyOwnedByYou
       end
 
+      lock!
+      reload
+      return if access_key && secret_key
```

Comment:
What do you think about moving this check to the beginning of the transaction so we don’t send multiple create bucket requests?